content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+
+## Building a reactive system
+
+Reactive systems are designed around the principle of data-driven reactivity. This is when signals and their dependents, effects, maintain a relationship. This relationship ensures a seamless propagation of data changes throughout the system, guaranteeing its responsiveness and real-time updates.
+
+### 1. Initialization
+
+The start of a reactive system begins with the setup of signals. During this phase, signals are initialized with their respective starting values. These initialized signals serve as the primary data sources that the rest of the system will rely on.
+
+### 2. Subscription
+
+Once signals are established, the next step involves the effects indicating their dependencies. 
+In this phase, effects "subscribe" to specific signals, marking their interest and establishing a link. 
+This subscription keeps the effects aware of any changes to the signals they are tied to.
+
+### 3. Notification
+
+During the notification phase, any alteration in a signal's value sets off a chain reaction. 
+The altered signal proactively notifies all its subscribed effects about the modification. 
+Upon receiving this notification, these effects then update or activate as necessary to keep the system's state current and synchronized with the latest data changes.
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity introduces a layer of flexibility and adaptability. 
+Unlike it synchronous counterpart, the updates to the effects are not instantaneous. 
+Instead, they may be delayed or could wait for the completion of an external event or task.
+
+This approach is useful in scenarios where an effect depends on multiple signals and when updating one signal before another may result a data inconsistency. 
+In such a scenario, [`batch`](/reference/batch) can be used to delay triggering effects so the effect runs after both signals have been updated.
+
+## Efficient data management
+
+### Memo
+
+Memo, short for memoization, is a technique that is used in reactive systems to enhance performance. 
+
+In the context of reactivity, memos become valuable with their ability to selectively recalculate based on changes in specific signals. 
+This means that a computed value or function can intelligently determine when to recalculate based on the signals it depends on. 
+
```

Comment:
It may be worth noting to that a memo is baaasically an effect that returns a signal to its return value. Like if one says that they are 95% the same thing then the whole system is easier to understand for the reader, imo.